src/provider/model/Storage.ts

```diff
@@ -1,13 +1,95 @@
 import localforage from 'localforage';
+import {injectable} from 'inversify';
 
-export const settingsStorage = localforage.createInstance({
-    driver: localforage.INDEXEDDB,
-    name: 'notifications',
-    storeName: 'settings'
-});
-
-export const notificationStorage = localforage.createInstance({
-    driver: localforage.INDEXEDDB,
-    name: 'notifications',
-    storeName: 'notifications'
-});
+import {AsyncInit} from '../controller/AsyncInit';
+
+/**
+ * List of available storages.
+ */
+export const enum StorageMap {
+    NOTIFICATIONS = 'notifications',
+    SETTINGS = 'settings'
+}
+
+@injectable()
+export class Storage extends AsyncInit {
+    /**
+     * This value should be updated any time changes are made to the notification shape.
+     * Must remain in number format.  LocalForage does not support semver.
+     */
+    public static readonly DATABASE_VERSION: number = 1.0;
+    private _storages: Map<string, LocalForage>;
+
+    constructor() {
+        super();
+        this._storages = new Map<string, LocalForage>();
+
+        [StorageMap.NOTIFICATIONS, StorageMap.SETTINGS].forEach(storeName => {
+            this.add({
+                driver: localforage.INDEXEDDB,
+                version: Storage.DATABASE_VERSION,
+                name: 'notifications',
+                storeName
+            });
+        });
+    }
+
+    protected async init(): Promise<void> {
+        await Promise.all([...this._storages.values()].map(storage => storage.ready()));
+
+        const settingsStorage = this.get(StorageMap.SETTINGS);
+
+        const storedVersion = await settingsStorage.getItem<number>('dbVersion');
+        const dbVersion = settingsStorage.config().version!;
+
+        if (!storedVersion) {
+            await this.initializeDatabase();
+        } else if (storedVersion < dbVersion) {
+            console.log(`Upgrading database from version ${storedVersion} to ${dbVersion}`);
+            await this.upgradeDatabase();
+        }
+    }
+
+    private add(options: LocalForageOptions): void {
```

Comment:
Would be tempted to inline this, since it's small, it's only called in one place, and if a second place called, that'd probably be a mistake

Reply:
I've moved it inline.  I was thinking more long time, but for now I guess it can live inline